android/src/main/java/com/optimove/reactnative/PushReceiver.java

```diff
@@ -31,6 +31,21 @@ protected void onPushOpened(Context context, PushMessage pushMessage) {
     PushReceiver.handlePushOpen(context, pushMessage, null);
   }
 
+  @Override
+  protected Intent getPushOpenActivityIntent(Context context, PushMessage pushMessage) {
+    Intent launchIntent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
+
+    if (null == launchIntent) {
+      return null;
+    }
+
+    launchIntent.putExtra(PushMessage.EXTRAS_KEY, pushMessage);
+    launchIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED);
```

Comment:
I think we can remove this and leave the launch method to be decided in the AndroidManifest.  For example, it is defined as singleTop in the example app

Reply:
leaving it to consumers gives more flexibility, but also means it has to be set correctly. We probably dont want a default (creates new activity) or singleTop (creates new if launch activity is not on top), but rather smth like "singleTask". having it as flags gives you sensible, consistent behaviour without doing any further work. Aso potentially more fine-grained control, like activity reparenting. not 100% which is better

Reply:
Afaik, RN creates a single activity in Android and its launch mode is singleTask, which is ok for this bug. I guess by consistent you mean that we have full control over it, which raises a question - if the user changed it for whatever reason, why overriding it? In any case, those lines cover our use case so we can keep them

Reply:
i didnt know RN adds singleTask by default, then y, these flags shall be ~same. Ideally would let consumers to override everything 

k, let's remove flags and rely on launch mode